lib/unimatrix/regent_sdk/email_settings_request.rb

```diff
@@ -15,16 +10,23 @@ def initialize( realm_uuid, access_token )
       end
       
       def retrieve
-        uri = URI( REGENT_URL + '/realms/' + @realm_uuid + '?access_token=' + @access_token + '&include[settings]' )
-        response = Net::HTTP.get( uri )
-        all_realm_settings = JSON.parse( response )[ 'settings' ]
-        email_settings = all_realm_settings.select { | setting | setting[ 'name' ][ 0..5 ] == 'email_' }
-        email_settings.map do | setting | 
-          setting.keep_if { | key, value | [ 'name', 'content' ].include?( key ) } 
-          setting[ 'name' ] = setting[ 'name' ][ 6..-1 ]
-        end
+        request = Request.new( 
+                    '/realms/' + @realm_uuid, 
+                    { access_token: @access_token, include_settings: true } 
+                  )
+                  
+        response = request.get
         
-        email_settings
+        if response.success?
+          all_realm_settings = response.settings
+          if all_realm_settings.respond_to?( :select )
```

Comment:
```
settings = response.settings
if settings.is_a?( Array ) && settings.first.type_name && settings.first.type_name == 'setting'
```
A Hash also responds to select but we want to check that it's an array. Also want to check that the array contains setting and not error objects.